transport/socks5/stream_dialer_test.go

```diff
@@ -192,9 +194,15 @@ func TestConnectWithoutAuth(t *testing.T) {
 
 	_, err = client.DialStream(context.Background(), address)
 	require.NoError(t, err)
+	// Forcefully close the listener to stop the server
+	// Otherwise defer listener.Close() will never be called
+	listener.Close()
```

Comment:
I believe this won't close if the test fails earlier. Instead, defer a function that does both the close and wait.